app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -223,60 +166,48 @@ class ReportingRepositoryImpl(
         }
     }
 
-    private suspend fun uploadRevokeSuspicionInfo(): MessageType.Revoke.Suspicion {
-        return withContext(coroutineContext) {
-            val infectionMessages = infectionMessageDao.getSentInfectionMessagesByMessageType(MessageType.InfectionLevel.Yellow)
-                .map { message ->
-                    message.publicKey to InfectionMessageContent(
-                        MessageType.Revoke.Suspicion,
-                        message.timeStamp,
-                        message.uuid
-                    )
-                }
-
-            apiInteractor.setInfectionInfo(
-                ApiInfectionInfoRequest(
-                    tanUuid.safeMap(defaultValue = EMPTY_STRING),
-                    tanDataSubject.value.tan,
-                    encryptInfectionMessages(infectionMessages),
-                    personalDataSubject.value.asApiEntity(MessageType.Revoke.Suspicion.warningType)
-                )
+    private suspend fun uploadData(warningType: WarningType,
+        temporaryExposureKeys: List<TemporaryExposureKey>) {
+        apiInteractor.uploadInfectionData(
+            generateApiTracingKeys(temporaryExposureKeys, warningType),
+            contextInteractor.packageName,
+            warningType,
+            ApiVerificationPayload(
+                tanUuid.safeMap(defaultValue = EMPTY_STRING),
+                tanDataSubject.value.tan
             )
+        )
+    }
+
+    private fun generateApiTracingKeys(temporaryExposureKeys: List<TemporaryExposureKey>, warningType: WarningType): List<ApiTemporaryTracingKey> {
+        return temporaryExposureKeys.map {
+            ApiTemporaryTracingKeyConverter.convert(it, warningType)
```

Comment:
please avoid static functions, have you considered an extension function like `temporaryExposreKeys.map{ it.convert(warningType)}` instead?